de.dlr.sc.virsat.team.ui/src/de/dlr/sc/virsat/team/ui/handler/git/GitCommitHandler.java

```diff
@@ -35,8 +37,28 @@ protected IVirSatVersionControlBackend createVersionControlBackend() {
 		return new VirSatGitVersionControlBackend(new EGitCredentialsProvider());
 	}
 	
+	@Override
+	protected void doUpdate(IProject project, IProgressMonitor monitor) throws Exception {
+		super.doUpdate(project, monitor);
+		
+		VirSatGitVersionControlBackend gitBackend = (VirSatGitVersionControlBackend) backend;
+		PullResult pullResult = gitBackend.getLastPullResult();
+		
+		Repository gitRepository = RepositoryMapping.getMapping(project).getRepository();
+
+		// Build the dialog in the UI threads
+		Display.getDefault().asyncExec(() -> {
+			PullResultDialog pullResultDialog = 
+					new PullResultDialog(Display.getDefault().getActiveShell(), gitRepository, pullResult);
+			pullResultDialog.open();
+		});
+	}
+	
 	@Override
 	protected void doCommit(IProject project, String message, IProgressMonitor monitor) throws Exception {
+		
+        // Update before committing
+        updateBeforeCommit(project, monitor);

```

Comment:
I think if you implement it here, it only works for git not SVN... Maybe move the implementation to the AVersionControlCommitHandler?

Reply:
I agree to that